### PR TITLE
feat: don't fetch node binaries if they are installed by package manager

### PR DESCRIPTION
**Pull request by darmalovan**
- [ ] Ready for review
- [ ] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?

We want to remove duplication of vulnerabilities for the case
that node installed by a package manager.

### CODE REVIEW

File: lib/analyzer/binaries-analyzer.ts [hunk 1]
```diff
@@ -7,15 +7,23 @@ export {
...
   return {
     Image: targetImage,
     AnalyzeType: 'binaries',
     Analysis: binaries,
   };
 }
 
+function installedByPackageManager(installedPackages: string[]): boolean {
+  return installedPackages
+    .filter((pkg) => pkg === 'node' || pkg === 'nodejs').length > 0;
```
Comment (robcresswell) [inline lib/analyzer/binaries-analyzer.ts]: Is there a way we can compare the results of the binary and system results so we don't need to keep manually updating this function?

Something like, get the binaries, then filter anything that is in the OS deps list

Comment (darmalovan) [reply lib/analyzer/binaries-analyzer.ts]: Let me work it out...

Comment (robcresswell) [reply lib/analyzer/binaries-analyzer.ts]: :crown:

**Comment by darmalovan** [comment]
@robcresswell LMK if this is what you thought about

**Pull request merged**